src/EventStore.Core/Messaging/ContinuationEnvelope.cs

```diff
@@ -36,4 +36,22 @@ public class ContinuationEnvelope : IEnvelope {
 			catch (OperationCanceledException) {}
 		}
 	}
+
+	public class AsyncTaskEnvelope : IEnvelope {
+		private readonly Func<Message, CancellationToken, Task> _onMessage;
+		private readonly CancellationToken _cancellationToken;
+
+		public AsyncTaskEnvelope(Func<Message, CancellationToken, Task> onMessage, CancellationToken cancellationToken) {
+			_onMessage = onMessage;
+			_cancellationToken = cancellationToken;
+		}
+
+		public void ReplyWith<T>(T message) where T : Message {
+			try {
+				_onMessage(message, _cancellationToken);
```

Comment:
depending on the implementation of _onMessage, the OperationCancelled exception might be caught here or it might end up on the Task (according to whether it is thrown synchronously or not). might be worth mentioned that in a comment so that it doesn't look like its something that might have been missed

Reply:
added, thanks